### DISCUSSION TITLE
Getting KeyError when I try to start the Remote webpage

### DISCUSSION THREAD

**Discussion by tmcafee52** [GerryDazoo/Slinger#100] [Q&A] [answered]
Thanks for all your effort on this.  I was pretty devastated thinking that my Slingbox wasn't going to work after November.  

I'm running the server on Windows 10 and can now get the video and audio using VLC on my local network and remotely using VLC on an Android phone.  However, on both, I try to call up the Remote webpage and am getting a number of KeyError's.  

I have Python 3.10.7 installed as a standalone and am running the sling_player.py file via the Python IDLE platform.  I've also tried just running the sling_player.exe without using Python IDLE.  I installed Flask by running the command "pip install flask" from the command prompt.  I then ran the commands "python" and then "import flask".

I have tried including the remote.txt in the config file and not including it.  

I haven't used Python before so I'm not sure what I'm doing and would greatly appreciate some help.

The output with the error messages are:
==================== RESTART: C:\Slingbox\slingbox_server.py ===================
Version : 3.07 Running on Windows-10-10.0.19044-SP0 pid= 7056
Using config file config.ini
Connection Manager Running on port 5201 with 10 max streams....
Streamer Running: starting up on  port 5201 
10 config.ini 5201
Checking for slingbox at ('192.168.1.254', 5201)
('192.168.1.254', 5201) OK
Using slingbox at  ('192.168.1.254', 5201)
Streamer:  tmcafee4_350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server'
 * Debug mode: on
[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5202
 * Running on http://192.168.1.100:5202
[33mPress CTRL+C to quit[0m
09/28/2022, 14:11:53  Registering Slingboxes {'11095DA8514999DF69C2F38D10572DCE': 5201}
Getting external IP address
Redirector, my external IP address 50.113.57.43
Registering 11095DA8514999DF69C2F38D10572DCE:5201

Register Successful
192.168.1.100 - - [28/Sep/2022 14:12:05] "[35m[1mGET /Remote HTTP/1.1[0m" 500 -
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2548, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Slingbox\slingbox_server.py", line 1116, in index
    return render_template_string(remotes[port][0] % stati[port])
KeyError: 5202
192.168.1.100 - - [28/Sep/2022 14:12:05] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -
192.168.1.100 - - [28/Sep/2022 14:12:05] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -
192.168.1.100 - - [28/Sep/2022 14:12:06] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -
192.168.1.100 - - [28/Sep/2022 14:12:06] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -

**Comment by GerryDazoo** [comment]
Please supply all the output from the program. And the URL you are using to try to connect the webpage, I suspect you are not using the correct port number to connect.
http://192.168.1.100:5201/Remote
Also in what you posted I don’t see a streaming request. The remote will only work while your streaming video.
PS. I think using the same port number for your server as your slingbox is causing you much confusion.

**Comment by tmcafee52** [comment]
Thanks for responding to my question.  I have included the entire output at the bottom of this message.  I'm using http://50.113.57.43:5201/slingbox for the slingbox and http://50.113.57.43:5202/Remote for the remote.  

Output:

Python 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.

==================== RESTART: C:\Slingbox\slingbox_server.py ===================
Version : 3.07 Running on Windows-10-10.0.19044-SP0 pid= 1160
Using config file config.ini
Connection Manager Running on port 5201 with 10 max streams....
starting up on  port 5201
Streamer Running:  10 config.ini 5201
Checking for slingbox at ('192.168.1.254', 5201)
('192.168.1.254', 5201) OK
Using slingbox at  ('192.168.1.254', 5201)
Streamer:  tmcafee4_350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server'
 * Debug mode: on
[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5202
 * Running on http://192.168.1.100:5202
[33mPress CTRL+C to quit[0m
09/29/2022, 12:24:58  Registering Slingboxes {'11095DA8514999DF69C2F38D10572DCE': 5201}
Getting external IP address
Redirector, my external IP address 50.113.57.43
Registering 11095DA8514999DF69C2F38D10572DCE:5201


Register Successful
09/29/2022, 12:25:08   Streaming connection from ('107.127.24.51', 55547)


Slinginfo  "350/500/M1" pwrfIpdH3TYUXYf 12 30 192.168.1.254 5201 3072
tmcafee4_350 Starting Stream for  107.127.24.51:55547
Connecting... ('192.168.1.254', 5201) Control
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.254', 5201) Stream
tmcafee4_350 Stream started at 09/29/2022, 12:25:13  744 0
....09/29/2022, 12:26:03  tmcafee4_350 Stream Terminated for  107.127.24.51:55547 [WinError 10054] An existing connection was forcibly closed by the remote host
tmcafee4_350 Shutting down connections
Streamer:  tmcafee4_350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
107.127.24.49 - - [29/Sep/2022 12:27:30] "[35m[1mGET /Remote HTTP/1.1[0m" 500 -
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2548, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Program Files\Python310\lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Slingbox\slingbox_server.py", line 1116, in index
    return render_template_string(remotes[port][0] % stati[port])
KeyError: 5202
107.127.24.49 - - [29/Sep/2022 12:27:30] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -
107.127.24.49 - - [29/Sep/2022 12:27:31] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -
107.127.24.49 - - [29/Sep/2022 12:27:31] "[36mGET /Remote?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -

**Comment by GerryDazoo** [comment]
Use http://50.113.57.43:5201/Remote for remote.

Make sure the router at 50.113.57.43 has a port forwarding rule to access the slingbox_server host 

 

ssage ID: ***@***.*** ***@***.***> >

**Comment by tmcafee52** [comment]
![netgear_port_fwd_slingbox](https://user-images.githubusercontent.com/112828368/193168020-d0d83e80-9f63-4605-923f-b882be0b3809.JPG)

I have 2 port forwarding entries in my router.  They are 5201 for the slingbox and 5202 for the remote.  They both forward to the slingbox server at 192.168.1.100.

Thanks

**Comment by GerryDazoo** [comment, marked as answer]
Again I will say. It’s port 5201 for the remote. DO NOT use port 5202.  http://192.168.1.100:5201/Remote

**Reply by tmcafee52** [reply]
Sorry that I got stuck on using the wrong port.  It's working now.  Thanks again for all your work on the slingbox and with helping me to resolve my problem.  You're the best.
